### PR TITLE
Update wrapper interface

### PR DESCRIPTION
**Pull request by roigcarlo**
This build updated KratosWrappers (CFD and DEM) to work with the current version of SimPhoNy common.

TODO:
[ ] Tests for KratosWrapper.py should be updated since old tests are now handled by cuds.

### CODE REVIEW

**Comment by roigcarlo** [comment]
@stefanoborini This PR should allow KratosWrappers to work with the last SimPohNy changes. It also adapt the wrappers to the new cuds methods.

I still have to finish some testing and probably some minor changes in the DEM Wrapper but they can be moved to another PR if you need something working ASAP.

Would you prefer to merge this and have the other changes in a different branch or is it better if I work directly here?

Please also notice that the setup.py versions are unchanged. Should we have the final tag for the release I will update it and set it as the min version for simphony-common

**Comment by stefanoborini** [comment]
Feel free to add the tests. I'm finalising other things on the framework/liggghts side, so I am not ready yet.

I'll do a release myself when everything is green, I'll have to use the commit ids, verify framework, then tag.

**Comment by roigcarlo** [comment]
Great thx

**Comment by stefanoborini** [comment]
@roigcarlo This is going to work against simphony-common master, but not on 0.4.0 (which is the version we need that doesn't change the interface, previously known as 413eb6f5683c4733b943c300c3192265c79ac26b ). See https://github.com/simphony/simphony-kratos/pull/28

**Comment by roigcarlo** [comment]
@stefanoborini Ok, I will leave this open and make another branch to generate a version that works with the 0.4 release.

**Comment by stefanoborini** [comment]
Thanks! The only major difference I suspect is the change in interface from get_something to get.
So if you extract everything but those changes, it should work, and then reintroduce them in a separate PR.

**Comment by roigcarlo** [comment]
@stefanoborini This should do the trick at least for now. There is a mechanism in Kratos that adds some restrictions on how to imports work. Probably that was the issue.
That said, I probably have to dig a little bit more to fully understand what is happening...

**Comment by stefanoborini** [comment]
@roigcarlo thanks. I'll review it just to get an idea later today and proceed to merge it.

File: setup.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 with open('README.rst', 'r') as readme:
     README_TEXT = readme.read()
 
-VERSION = '0.2.1.dev0'
+VERSION = '0.5.dev0'
```
Comment (stefanoborini) [inline setup.py:8]: Version should probably just become 0.3 here.

File: simkratos/CFD/kratos_CFD_wrapper.py [hunk 2]
```diff
@@ -398,66 +406,69 @@ def importKratosConditions(self, src, dst, group):
...
+    def importKratosDof(self, src, dst, group):
 
-        mesh_prop = Properties(group)
+        bc = self.get_cuds().get(src.data[CUBA.CONDITION])
 
+        mesh_prop = Properties(group)
         mesh_prop.SetValue(IS_SLIP, 0)
 
-        if self.BC[CUBA.PRESSURE][mesh_name] == 'empty':
+        print("Applying DOF to mesh", group, bc, src.data[CUBA.CONDITION])
```
Comment (stefanoborini) [inline simkratos/CFD/kratos_CFD_wrapper.py]: should we leave this? Don't know... maybe it's ok.

File: simkratos/CFD/kratos_CFD_wrapper.py [hunk 3]
```diff
@@ -398,66 +406,69 @@ def importKratosConditions(self, src, dst, group):
...
+
+        if CUBA.PRESSURE not in bc.data:
             mesh_prop.SetValue(IMPOSED_PRESSURE, 0)
         else:
+            print(bc.data[CUBA.PRESSURE])
```
Comment (stefanoborini) [inline simkratos/CFD/kratos_CFD_wrapper.py]: same

File: simkratos/CFD/kratos_CFD_wrapper.py [hunk 4]
```diff
@@ -398,66 +406,69 @@ def importKratosConditions(self, src, dst, group):
...
-            )
-
-            for node in self.fluid_model_part.GetNodes(group):
-                node.Fix(VELOCITY_X)
-                node.Fix(VELOCITY_Y)
-                node.Fix(VELOCITY_Z)
-                node.SetValue(
-                    VELOCITY_X,
-                    self.BC[CUBA.VELOCITY][mesh_name][0]
+            print(bc.data[CUBA.VELOCITY])
```
Comment (stefanoborini) [inline simkratos/CFD/kratos_CFD_wrapper.py]: same

File: simkratos/CFD/kratos_CFD_wrapper.py [hunk 5]
```diff
@@ -478,60 +489,54 @@ def initialize(self):
...
+                mesh, self.fluid_model_part, group
             )
 
+            meshDict[mesh.name] = meshNumber
+
             properties.append(mesh_prop)
+            meshNumber += 1
 
         self.fluid_model_part.SetProperties(properties)
+        print(self.fluid_model_part)
```
Comment (stefanoborini) [inline simkratos/CFD/kratos_CFD_wrapper.py]: same

File: simkratos/DEM/kratos_DEM_wrapper.py [hunk 6]
```diff
@@ -604,30 +615,35 @@ def initialize(self):
...
                 particles,
                 self.spheres_model_part,
                 group
             )
 
-        # for mesh_name in solid_meshes:
+            meshDict[mesh.name] = meshNumber
+            meshNumber += 1
+
+        # for mesh in cuds.iter(ABCMesh):
```
Comment (stefanoborini) [inline simkratos/DEM/kratos_DEM_wrapper.py]: Remove commented lines..

File: setup.py (hunk 1 above)
Comment (roigcarlo) [reply setup.py:8]: I would prefer jump to 0.5.
If I remember correctly we agreed (or at least I understood) that it was desirable that both wrappers and core maintain the same 0.X version.

**Comment by stefanoborini** [comment]
@roigcarlo good to merge?

**Comment by roigcarlo** [comment]
:+1:

**Comment by stefanoborini** [comment]
@roigcarlo thanks!!

**Pull request merged**